src/reducers/expressions.js

```diff
@@ -96,12 +102,18 @@ function deleteExpression(state: State, input: string) {
 
 type OuterState = { expressions: Record<ExpressionState> };
 
-function getExpressions(state: OuterState) {
-  return state.expressions.get("expressions");
+function getExpressions(state: OuterState, visible: boolean = false) {
+  const expressions = state.expressions.get("expressions");
+  return visible ? expressions.filter(e => e.visible) : expressions;
+}
```

Comment:
This is a minor preference, but I think we could likely split this into two selectors: `getExpressions` and `getVisibleExpressions`